### PR TITLE
fix: don't cut off a long runUrl in console

### PR DESCRIPTION
**Pull request by mirobo**
renderTables will cut off longer urls which "renders" the url unusable

### User facing changelog
After a recorded run, a long runUrl would cut off which made it unusable
 Recorded Run: https://......................../run/3806241066d18ec315b3a8f…    ****

### How has the user experience changed?
Whenever the runUrl is displayed in Gitlab CI log, the URL to the run will be clickable and it will open the according run

See this incomplete URL:
image

### PR Tasks

- [x] Have tests been added/updated?
- [ ] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

**Comment by lmiller1990** [comment]
Thanks for the PR! Looks like we've still got some fails. If you go into `system-tests` and run `SNAPSHOT_UPDATE=1 yarn test reocrd_spec` it should run all the specs and update the snapshots for you.

**Review by lmiller1990 — COMMENTED** [review]
Cool, just need to get CI green - see my comment.

**Comment by mirobo** [comment]
> Thanks for the PR! Looks like we've still got some fails. If you go into `system-tests` and run `SNAPSHOT_UPDATE=1 yarn test reocrd_spec` it should run all the specs and update the snapshots for you.

Thanks for the hint. It was extremly hard to find some sort of hint in the CI runs/logs that would point to the error. So I just searched for the string "Recorded Run"

I have to pray that it runs on my private computer though. I got quite a few weird error messages.. probably because I don't have Linux Subsystem installed on Windows (at some point it tries to execute a bash command?). When I ran yarn or read the README I couldn't find anything specific about the prerequisites for the development setup. I just installed yarn and ran the yarn command.

And the examples of running tests are probably wrong.. 
starting in the directory "system-test", the command "yarn test test/record_spec.js" won't work but "yarn test record_spec.js" does at least find the file. Not quite sure if this different on Mac and Windows..

I also don't know if the tests would be run with Cypress that includes my changes (do I need to build a Cypress.exe with "binary-build"?)

**Comment by lmiller1990** [comment]
> I got quite a few weird error messages.. probably because I don't have Linux Subsystem installed on Windows (at some point it tries to execute a bash command?). When I ran yarn or read the README I couldn't find anything specific about the prerequisites for the development setup. I just installed yarn and ran the yarn command.

For linux/mac should be just `yarn install`. I've done development on windows with git bash too, which works okay.

>And the examples of running tests are probably wrong..
> starting in the directory "system-test", the command "yarn test test/record_spec.js" won't work but "yarn test record_spec.js" does at least find the file. Not quite sure if this different on Mac and Windows.

Yeah this seems incorrect - we should update the docs. The system test looks inside `test` w/ a regexp so just `yarn test record` would work.

> I also don't know if the tests would be run with Cypress that includes my changes (do I need to build a Cypress.exe with "binary-build"?)

Most tests run against the source code, not the bundled binary. We do have binary tests, but running those locally (except on linux) is a little annoying, so I generally recommend people just push to CI for those.

**Comment by lmiller1990** [comment]
Something still isn't right:

image

I think we still need the `table` calls. I will look.

**Comment by lmiller1990** [comment]
I think we can simplify this, pushed a commit, let's see how it goes.

**Comment by lmiller1990** [comment]
Ok got this working but we don't actually have a test for a really long URL. I'd like to add one. @mirobo please wait a little, or if you'd like to try and add one, that'd be great, too. I have no idea how to approach adding this, though.

If you'd like to grab my commits and at least verify it's still working as you expect, that'd be fine, too.

**Comment by mirobo** [comment]
Thanks for your additions/fixes! I'll check for a long real-life example later..

I fixed some other issues in package.json and README.MD.. I wonder what the CI builds say to this :D

> > I got quite a few weird error messages.. probably because I don't have Linux Subsystem installed on Windows (at some point it tries to execute a bash command?). When I ran yarn or read the README I couldn't find anything specific about the prerequisites for the development setup. I just installed yarn and ran the yarn command.
> 
> For linux/mac should be just `yarn install`. I've done development on windows with git bash too, which works okay.

First I needed to install Yarn globally for Windows and I needed to add the yarn binary dir (i.e. C:\Program Files (x86)\Yarn\bin\) to the PATH env variable.

Building the binary is somehow required for the system-test. When I run the a system-test:

```
listening on port: 1234
listening on port: 3131
📦 Found package.json for project e2e.
📦 No dependencies found, skipping dep-installer steps
spawn C:\Dev\mirobo\cypress\packages\electron\dist\Cypress\Cypress.exe ENOENT
Error: spawn C:\Dev\mirobo\cypress\packages\electron\dist\Cypress\Cypress.exe ENOENT
    at Process.ChildProcess._handle.onexit (node:internal/child_process:282:19)
    at onErrorNT (node:internal/child_process:477:16)
    at processTicksAndRejections (node:internal/process/task_queues:83:21)
      1) passes
```

But I got this error when running "yarn binary-build":
```
$ C:\Dev\mirobo\cypress\node_modules\.bin\lerna run build-prod --ignore cli --concurrency 1
/usr/bin/bash: C:Devmirobocypressnode_modules.binlerna: command not found
```

Then I started to fiddle around with installing Lerna via NPM globally and also via Yarn globally. I was loosing my mind, I tried stuff for over an hour, googled at least half an hour and it still failed with the same error. It was a real pain!

In the end I uninstalled Lerna globally via NPM and installed it globally via Yarn. And then suddenly it started... but the joy was not for long.

"yarn binary-build" got further and produced a Cypress binary in "C:\Dev\mirobo\cypress\packages\electron\dist".  But still with the following error..
It just doesn't work out of the box.. it's a real pain.. it shouldn't be so hard to set this up on a new PC.. The prerequisites are totally ambiguous.

```
3:09:31 All paths are short enough (29590 checked) win32
in build folder C:\Users\root\AppData\Local\Temp\cypress-build\win32\build\win-unpacked
🔥 deploy error
Error: Command failed with ENOENT: ls -la C:\Users\root\AppData\Local\Temp\cypress-build\win32\build\win-unpacked
spawn ls ENOENT
Der Befehl "ls" ist entweder falsch geschrieben oder
konnte nicht gefunden werden.
    at notFoundError (C:\Dev\mirobo\cypress\node_modules\cross-spawn\lib\enoent.js:6:26)
    at verifyENOENT (C:\Dev\mirobo\cypress\node_modules\cross-spawn\lib\enoent.js:40:16)
    at ChildProcess.cp.emit (C:\Dev\mirobo\cypress\node_modules\cross-spawn\lib\enoent.js:27:25)
    at Process.ChildProcess._handle.onexit (node:internal/child_process:290:12) {
  code: 'ENOENT',
  errno: 'ENOENT',
  syscall: 'spawn ls',
  path: 'ls',
  spawnargs: [
    '-la',
    'C:\\Users\\root\\AppData\\Local\\Temp\\cypress-build\\win32\\build\\win-unpacked'
  ],
  originalMessage: 'spawn ls ENOENT',
  shortMessage: 'Command failed with ENOENT: ls -la C:\\Users\\root\\AppData\\Local\\Temp\\cypress-build\\win32\\build\\win-unpacked\n' +
    'spawn ls ENOENT',
  command: 'ls -la C:\\Users\\root\\AppData\\Local\\Temp\\cypress-build\\win32\\build\\win-unpacked',
  exitCode: undefined,
  signal: undefined,
  signalDescription: undefined,
  stdout: '',
  stderr: 'Der Befehl "ls" ist entweder falsch geschrieben oder\r\n' +
    'konnte nicht gefunden werden.',
  failed: true,
  timedOut: false,
  isCanceled: false,
  killed: false
}
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```


Then I try to run the record_spec.js.. not a single test completes which may have to do with the fact that Cypress 10 gets hung on my PC after some usage. Need to gather some memory dumps for this..

**Comment by lmiller1990** [comment]
You should not need to build the binary for most system tests. Some run with the binary, but the majority do not.

Are you just doing 

```sh
yarn
cd system-tests
yarn test record
```

This should execute the record_spec.js against the real code - NOT the binary. I can fire up windows later to verify this, you might have found a bug, but I'm really surprised since I'm sure people have developed Cypress on windows in the past.

**Comment by lmiller1990** [comment]
I'm sorry I couldn't look at this today, I will try to give you another review and also check the state of windows + system tests before the end of the week.

**Comment by lmiller1990** [comment]
I added a test.

```
 lib/cypress
    --record
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------
                                                  lib/cypress / --record / does not truncate a really long dashboard url                                                  
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------

====================================================================================================

  (Run Starting)

  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ Cypress:    10.3.0                                                                             │
  │ Browser:    Electron 99 (headless)                                                             │
  │ Specs:      1 found (app.cy.js)                                                                │
  │ Searched:   cypress/e2e/**/*.cy.{js,jsx,ts,tsx}                                                │
  │ Params:     Tag: false, Group: electron-smoke-tests, Parallel: false                           │
  │ Run URL:    http://dashboard.cypress.io/this-is-a-long-long-long-long-long-long-long-long-long │
  │             -long-long-long-long-long-long-long-long-long-long-long-long-long-long-long-long-l │
  │             ong-long-long-long-long-long-long-long-long-long-long-long-long-long-long-long-lon │
  │             g-long-long-long-long-long-long-long-long-url                                      │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘


────────────────────────────────────────────────────────────────────────────────────────────────────
                                                                                                    
  Running:  app.cy.js                                                                       (1 of 1)

====================================================================================================

  (Run Finished)


       Spec                                              Tests  Passing  Failing  Pending  Skipped  
  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ ✖                                             6ms        1        2        3        4        5 │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘
    ✖  1 of 1 failed (100%)                       6ms        1        2        3        4        5  


───────────────────────────────────────────────────────────────────────────────────────────────────────
                                                                                                       
  Recorded Run: http://dashboard.cypress.io/this-is-a-long-long-long-long-long-long-long-long-long-long-long-long-long-long-long-long-long-long-long-long-long-long-long-long-long-long-long-long-long-long-long-long-long-long-long-long-long-long-long-long-long-long-long-long-long-long-long-long-long-long-url

      ✓ does not truncate a really long dashboard url (588ms)


  1 passing (2s)

```

**Review by astone123 — APPROVED** [review]

File: packages/server/test/integration/cypress_spec.js [hunk 1]
```diff
@@ -99,6 +99,53 @@ const snapshotConsoleLogs = function (name) {
   return snapshot(name, stripAnsi(args))
 }
 
+function mockEE () {
```
Comment (lmiller1990) [inline packages/server/test/integration/cypress_spec.js:102]: Just moved duplicated code to this function. The implementation was not changed.

**Comment by lmiller1990** [comment]
Looks good. I'll get one more :eyes: from the e2e team, this should be able to make it into the next release. Thanks @mirobo!

**Comment by mirobo** [comment]
> Looks good. I'll get one more 👀 from the e2e team, this should be able to make it into the next release. Thanks @mirobo!

Thanks! I've checked the failing tests but I really, really couldn't figure out what my changes would have to do with those failures. I guess they were present on the develop branch anyway?

**Pull request merged**